pallets/stateful-storage/src/lib.rs

```diff
@@ -110,35 +144,101 @@ pub mod pallet {
 
 		/// Page size exceeds max allowable page size
 		PageExceedsMaxPageSizeBytes,
+
+		/// Invalid SchemaId or Schema not found
+		InvalidSchemaId,
+
+		/// Invalid Message Source Account
+		InvalidMessageSourceAccount,
+
+		/// UnAuthorizedDelegate
+		UnAuthorizedDelegate,
+
+		/// Corrupted State
+		CorruptedState,
+
+		/// Invalid item action
+		InvalidItemAction,
 	}
 
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
-		ItemAppended,
-		ItemRemoved,
-		PageAppended,
-		PageUpdated,
-		PageRemoved,
+		PageUpdated { msa_id: MessageSourceId, schema_id: SchemaId },
```

Comment:
Maybe add `page_id: PageId[]` to the event? Or not worth it?

Reply:
I think we will definitely want page number for paginated ones. maybe an `Optional<PageNumber>`